Project1/spaceInvaders/spaceInvaders.pde

```diff
@@ -0,0 +1,126 @@
+// Space
+
+int UFOK_SZAMA = 5;
+float ufoPozicioX[] = new float[UFOK_SZAMA];
+float ufoPozicioY[] = new float[UFOK_SZAMA];
+float ufoSzelesseg = 30;
+float ufoMagassag = 20;
+float ufoEltolasOldalra;
+float UFO_SEBESSEG = 5;
+boolean ufoMozogLe = false;
+
+float balMargo;
+float jobbMargo;
+
+float fegyverPozicioX;
+
+float mousePrevX;
+
+void setup() {
+  size(1024, 600);
+
+  for (int idx = 0; idx < ufoPozicioX.length; ++idx) {
+    ufoPozicioX[idx] = width / 2 - (ufoPozicioX.length/2 * (ufoSzelesseg * 2.5)) + (ufoSzelesseg * 2.5 * idx);  
+    ufoPozicioY[idx] = 100;
+  }
+
+  ufoEltolasOldalra = -UFO_SEBESSEG;
+  ufoMozogLe = false;
+
+  balMargo = width / 10;
+  jobbMargo = width - width / 10;
+
+  fegyverPozicioX = (width + 40)/2;
+  mousePrevX = 0;
+}
+
+void fegyverRajzol() {
+  stroke(#777777);
+  fill(#777777);
+  arc(fegyverPozicioX, height - 30, 40, 40, PI, TWO_PI);
+  rect(fegyverPozicioX - 1.5, height - 60, 3, 10);
+}
+
+void LezerLoves() {
+  stroke(#990000);
+  fill(#990000);
+  rect(fegyverPozicioX - 1.5, 0, 3, height - 60);
+}
+
+void ufoRajzol() {
+  stroke(#F0DE11);
+  fill(#F0DE11);
+  for (int idx = 0; idx < ufoPozicioX.length; ++idx) {
+    if (0 < ufoPozicioX[idx])
+      ellipse(ufoPozicioX[idx], ufoPozicioY[idx], ufoSzelesseg, ufoMagassag);
+  }
+}
+
+void ufoMozgatOldalra() {
+  for (int idx = 0; idx < ufoPozicioX.length; ++idx) {
+    if (0 > ufoPozicioX[idx])
+      continue;
+    ufoPozicioX[idx] += ufoEltolasOldalra;
+    if (balMargo >= ufoPozicioX[idx]) {
+      ufoEltolasOldalra = UFO_SEBESSEG;
+      ufoMozogLe = true;
+      break;
+    }
+    if (jobbMargo <= ufoPozicioX[idx]) {
+      ufoEltolasOldalra = -UFO_SEBESSEG;
+      ufoMozogLe = true;
+      break;
+    }
+  }
+}
+
+void ufoMozgatLefele() {
+  if (!ufoMozogLe)
+    return;
+  ufoMozogLe = false;
+  for (int idx = 0; idx < ufoPozicioY.length; ++idx) {
+    if (0 > ufoPozicioY[idx])
+      continue;
```

Comment:
Köszönöm Ákos az észrevételeidet!
Kijavítom őket, viszont az a helyzet, hogy holnaptól elutazom és
legkorábban hétfőn tudom kitenni a javításokat. Megfelel ez így neked?

Üdv,
Gergő

On Thu, Mar 12, 2015 at 4:07 PM, Akos Zsolt Hochrein <
notifications@github.com> wrote:

> In Project1/spaceInvaders/spaceInvaders.pde
> <https://github.com/prezi/skool-curriculum-highschoolers/pull/26#discussion_r26309519>
> :
>
> > +    }
> > +    if (jobbMargo <= ufoPozicioX[idx]) {
> > +      ufoEltolasOldalra = -UFO_SEBESSEG;
> > +      ufoMozogLe = true;
> > +      break;
> > +    }
> > +  }
> > +}
> > +
> > +void ufoMozgatLefele() {
> > +  if (!ufoMozogLe)
> > +    return;
> > +  ufoMozogLe = false;
> > +  for (int idx = 0; idx < ufoPozicioY.length; ++idx) {
> > +    if (0 > ufoPozicioY[idx])
> > +      continue;
>
> continue nem tananyag.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/prezi/skool-curriculum-highschoolers/pull/26/files#r26309519>
> .
>



-- 

*Gergely Mancz*
hu.linkedin.com/in/gergelymancz <http://www.linkedin.com/in/gergelymancz>
mobile: +36 30 339 7897
e-mail: gergelymancz@gmail.com